### PR TITLE
Add Prompt.Create option

### PR DESCRIPTION
**Pull request by pmaytak**
Issue #2463

Also updates Xamarin Dev App and DesktopTestApp

### CODE REVIEW

**Comment by bgavrilMS** [comment]
LGTM

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 1]
```diff
@@ -4,30 +4,30 @@
...
+    /// Structure containing static members that you can use to specify how the interactive overrides
+    /// of AcquireTokenAsync in <see cref="IPublicClientApplication" /> should prompt the user.
     /// </summary>
     public struct Prompt
     {
         /// <summary>
-        ///     AcquireToken will send <c>prompt=select_account</c> to Azure AD's authorize endpoint
-        ///     which would present to the user a list of accounts from which one can be selected for
-        ///     authentication.
+        /// AcquireToken will send <c>prompt=select_account</c> to Azure AD's authorize endpoint
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:13]:
```suggestion
        /// AcquireToken will send <c>prompt=select_account</c> to the authorization server's authorize endpoint.
```

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 2]
```diff
@@ -4,30 +4,30 @@
...
         /// <remarks>
         /// This is the default prompt for backwards compatibility reasons. Please use <see cref="Prompt.NoPrompt"/> for the best user experience.
         /// </remarks>
         public static readonly Prompt SelectAccount = new Prompt("select_account");
 
         /// <summary>
-        ///     The user will be prompted for credentials by the service. It is achieved
-        ///     by sending <c>prompt=login</c> to the Azure AD service.
+        /// The user will be prompted for credentials by the service. It is achieved
+        /// by sending <c>prompt=login</c> to the Azure AD service.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:24]:
```suggestion
        /// by sending <c>prompt=login</c> to the authorize endpoint.
```

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 3]
```diff
@@ -4,30 +4,30 @@
...
         /// </summary>
         public static readonly Prompt ForceLogin = new Prompt("login");
 
         /// <summary>
-        ///     The user will be prompted to consent, even if consent was granted before. It is achieved
-        ///     by sending <c>prompt=consent</c> to Azure AD.
+        /// The user will be prompted to consent, even if consent was granted before. It is achieved
+        /// by sending <c>prompt=consent</c> to Azure AD.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:30]:
```suggestion
        /// by sending <c>prompt=consent</c> to the authorization server's authorize endpoint.
```

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 4]
```diff
@@ -38,12 +38,18 @@ public struct Prompt
         /// <remarks>This is the recommended prompt</remarks>
         public static readonly Prompt NoPrompt = new Prompt("no_prompt");
 
+        /// <summary>
+        /// AcquireToken will send <c>prompt=create</c> to Azure AD's authorize endpoint
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:42]:
```suggestion
        /// AcquireToken will send <c>prompt=create</c> to the authorization server's authorize endpoint
```

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 5]
```diff
@@ -38,12 +38,18 @@ public struct Prompt
...
+        /// which would trigger a sign-up experience.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:43]:
```suggestion
        /// which would trigger a sign-up experience, used for External Identities. 
```

do we have an aka.ms link for this? https://docs.microsoft.com/en-us/azure/active-directory/external-identities/self-service-sign-up-user-flow @jmprieur

File: src/client/Microsoft.Identity.Client/Prompt.cs [hunk 6]
```diff
@@ -38,12 +38,18 @@ public struct Prompt
...
+
 #if DESKTOP || WINDOWS_APP
-/// <summary>
-/// Only available on .NET platform. AcquireToken will send <c>prompt=attempt_none</c> to
-/// Azure AD's authorize endpoint and the library will use a hidden webview (and its cookies) to authenticate the user.
-/// This can fail, and in that case a <see cref="MsalUiRequiredException"/> will be thrown.
-/// </summary>
+        /// <summary>
+        /// Only available on .NET platform. AcquireToken will send <c>prompt=attempt_none</c> to
+        /// Azure AD's authorize endpoint and the library will use a hidden WebView (and its cookies) to authenticate the user.
```
Comment (jennyf19) [inline src/client/Microsoft.Identity.Client/Prompt.cs:50]:
```suggestion
        /// the authorization server's authorize endpoint and the library will use a hidden WebView (and its cookies) to authenticate the user.
```

**Comment by jennyf19** [comment]
I think the summary statements should be more generic. Also, have you tested w/AAD and AAD B2C scenarios? Non-External Identities...what is the experience?

**Comment by jmprieur** [comment]
we don't have an aka.ms link yet. We should create it.

File: src/client/Microsoft.Identity.Client/Prompt.cs (hunk 5 above)
Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/Prompt.cs:43]: here: https://aka.ms/msal-net-prompt-create @jmprieur @pmaytak

**Review by bgavrilMS — APPROVED** [review]

**Comment by pmaytak** [comment]
@jmprieur @jennyf19 
Testing passing Prompt.Create with the [DesktopTestApp](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/tree/master/tests/devapps/DesktopTestApp):
1. For AAD tenant that doesn't support External Identities, it seems to act as no prompt, it logs it my MS account.
2. For B2C I get this error:
```
Error Code - invalid_request
Message - AADB2C90043: The prompt included in the request contains invalid values. Expected 'none', 'login', 'consent' or 'select_account'.
```
3. If I use the client ID and tenant ID from #2463, the sign up screen shows up correctly.
4. I followed the [directions](https://docs.microsoft.com/en-us/azure/active-directory/external-identities/self-service-sign-up-user-flow) and set up external identities in my own tenant. I get mixed behavior. If I start the up and immediately send Create, it still logs in my MS account. If I pass SelectAccount, login with an account from that tenant, then try to pass Create again - the sign up screen shows up correctly.

**Comment by bgavrilMS** [comment]
Thanks for testing in such detail @pmaytak - I think the PR can be merged now, since the new prompt is an opt-in feature, it isn't blocking anybody and the error message is good. We can follow up with B2C and AAD team separately.

**Comment by jmprieur** [comment]
@pmaytak : 
- did you try for 1) what happens if you were not already signed-in (maybe with a system browser)?
- For 2. the error is this is explicit. ow is it surfaced? as an MsalServiceException? 
- 4. Did you see if the prompt is sent when you start the app without doing the select account first? did you see what happens when you sign-out first?

**Comment by jennyf19** [comment]
@jmprieur do you have more info on the scenario and what is expected? Just curious here w/signing-in w/an active session...and, +1 to what you said when signing-out first, and then running the app w/prompt.create. 

I'll ping someone from b2c w/some questions about this as well.

**Comment by pmaytak** [comment]
- I only tried with the embedded browser and didn't try sign out. I did revoke sessions in the AAD tenant itself but the behavior is the same. 
- Yes, prompt create is always added correctly to the authorize URI.
- Yes, the B2C error is MsalServiceException.
- I can update the docs to mention that this is not for B2C.

**Comment by pmaytak** [comment]
Tried with the system browser. Made sure to sign out beforehand. Made the first request Prompt.Create and the sign up window showed correctly. I was able to sign up with an outlook email successfully. But then all following Prompt.Create requests used that cached account and didn't show the signup window.

**Comment by jmprieur** [comment]
@pmaytak thanks for checking. I think that's the expected experience. you would not sign-up without sign-in out
this is a sign-up / sign-in flow then.

**Pull request merged**